pkg/onchain/website/handler.go

```diff
@@ -38,6 +38,16 @@ func MassaTLDInterceptor(req *interceptor.Interceptor) *interceptor.Interceptor
 	massaIndex := strings.Index(req.Request.Host, ".massa")
 
 	if massaIndex > 0 && !strings.HasPrefix(req.Request.Host, "my.massa") {
+		// If the request is not https, redirect to https.
+		if req.Request.TLS == nil {
+			http.Redirect(
+				req.Writer,
+				req.Request,
+				"https://"+req.Request.Host+req.Request.URL.Path,
```

Comment:
here we can lost query params and anchor, can't you get the full URL and replace http:// by https://?

Reply:
> here we can lost query params and anchor
Good catch ! I will check if it's working well

> can't you get the full URL and replace http:// by https://?
No, I only have the `my.massa/thyra/...` for example. Not the protocol or other things like that.

Reply:
The empty `RawQuery` was due to my WebBrowser cache. 

It should be fixed with [cd4c87c](https://github.com/massalabs/thyra/pull/534/commits/cd4c87c7a4010ebc0bd058846e0c539558d3a40b)